src/models/Order.js

```diff
@@ -16,6 +16,13 @@ const OrderItemSchema = new mongoose.Schema({
   },
 });
 
+// Add a virtual property to populate the album with all its properties
+OrderItemSchema.virtual('fullAlbum').get(function () {
+  return this.populate('album').execPopulate().then((doc) => {
+    return doc.album;
+  });
+});
+
```

Comment:
I don't think this will work; I'm getting the following error when I try to access the virtual property (order.orderItems\[0\].fullAlbum):

```
`Error while processing the order: Error: Mongoose does not support calling populate() on nested docs. Instead of `doc.arr[0].populate("path")`, use `doc.populate("arr.0.path")`
```